### PR TITLE
[hi] Merge Hindi localization branch to main

### PR DESCRIPTION
**Pull request by anubha-v-ardhan**
### Merge `dev-1.22-hi.1` to `main`

Approvers and/or Reviewers for Hindi l10n:
@anubha-v-ardhan 
@divya-mohan0209 
@Garima-Negi 
@verma-kunal 

[Deploy preview](https://deploy-preview-29679--kubernetes-io-main-staging.netlify.app/hi/)

### CODE REVIEW

**Comment by sftim** [comment]
The preview won't include Hindi until https://github.com/kubernetes/website/pull/29674 (or an equivalent change) is merged.

**Comment by sftim** [comment]
Currently waiting on `content/hi/docs/concepts/overview/what-is-kubernetes.md` before this will preview OK.

**Comment by farukranabappy89** [comment]
solution and development

**Comment by sftim** [comment]
I'm looking at this. I can see that the site deploy to Netlify has failed; I'll look into that.

**Comment by sftim** [comment]
> `Error: Error building site: "/opt/build/repo/content/hi/docs/setup/production-environment/container-runtimes.md:46:6": closing tag for shortcode 'caution' does not match start tag`

That will need fixing, folks.

File: content/hi/docs/setup/production-environment/container-runtimes.md
Comment (Sea-n) [inline content/hi/docs/setup/production-environment/container-runtimes.md:46]: This caused build error.
```suggestion
{{< caution >}}
```

**Review by Sea-n — COMMENTED** [review]
Please refer to [Netlify build log](https://app.netlify.com/sites/kubernetes-io-main-staging/deploys/625195a29ad7fd0009c10ff4#L129), feel free to `/unhold` yourself when the issue is solved.

/hold

**Comment by mehabhalodiya** [comment]
We couldn't merge this as-is as there is a conflict that the automatic system does not know how to resolve.

@anubha-v-ardhan if you haven't done a rebase before, you really have 2 options:

1. Learn about Git rebasing. It's a great skill if you often work with source control; on the other hand, if you're mostly a writer or focus your skills in other areas than Git, it's probably not worth the effort (it's genuinely not that easy to learn the first time round).
2. Close this pull request, and open a new one based on the main branch that makes the same changes. In the new PR, mention that it's a replacement for PR https://github.com/kubernetes/website/pull/29679

Here are some references/tips for the following:

- [merge conflicts and rebasing](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#merge-conflicts-and-rebasing)
- [squashing commits](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits)

File: content/hi/docs/setup/production-environment/container-runtimes.md
Comment (anubha-v-ardhan) [reply content/hi/docs/setup/production-environment/container-runtimes.md]: Thank you @Sea-n :)

**Comment by anubha-v-ardhan** [comment]
>@anubha-v-ardhan if you haven't done a rebase before, you really have 2 options:

@Garima-Negi I'm comfortable with rebase but not sure if I have the right to rebase a branch on upstream repo. I would request someone with access to help solve the merge conflict.

**Comment by sftim** [comment]
Despite the message from Prow, this _doesn't_ need a rebase. The right fix here is to merge in changes from _main_ whilst resolving the merge conflict for `config.toml`.

**Comment by sftim** [comment]
@anubha-v-ardhan if you'd like help sorting out that merge conflict, ask in #sig-docs on Kubernetes Slack whether there is someone who can make time to pair with you on it. That'd be the best option if it works out, I think.

**Comment by sftim** [comment]
Site has previewed OK.
/hold cancel

**Comment by sftim** [comment]
However, I'd like to add a new hold. This PR adds a whole new localization; let's check in with SIG Docs leads (I'm one of the tech leads) before we merge this.

/hold

**Comment by Sea-n** [comment]
Hey @anubha-v-ardhan,

This PR have `needs-rebase` label for 2 months. Would you like to solve it so that we can move forward?

**Comment by Garima-Negi** [comment]
> Hey @anubha-v-ardhan,
> 
> This PR have `needs-rebase` label for 2 months. Would you like to solve it so that we can move forward?

working on this

**Comment by Sea-n** [comment]
Superseded by #35651
/close

**Comment by sftim** [comment]
/reopen
https://github.com/kubernetes/website/pull/35651 does not supersede this PR.

**Comment by sftim** [comment]
Hindi approvers: how do these changes look?

File: config.toml
Comment (rolfedh) [inline config.toml:33]: Will this change enable Hindi on the published website? Is that the intention of this PR?

File: config.toml
Comment (rolfedh) [inline config.toml:445]:
```suggestion
languageName = "हिन्दी (Hindi)"
```
- Main has been updated to follow the [language name standard](https://github.com/kubernetes/website/blob/main/content/en/docs/contribute/localization.md#modify-the-site-configuration). Also see #31657
- We should update this line in `dev-1.22-hi.1` so it is the same as `main`.

File: config.toml
Comment (rolfedh) [inline config.toml:453]: Should we update these lines in `dev-1.22-hi.1` so that they are the same as `main`?

File: config.toml
Comment (rolfedh) [reply config.toml:445]: I have opened an issue and PR to fix this in `dev-1.22-hi.1`:
- https://github.com/kubernetes/website/issues/35783
- https://github.com/kubernetes/website/pull/35784

File: config.toml
Comment (sftim) [reply config.toml:33]: Yes; merging this PR will launch Hindi.

Comment (rolfedh) [reply config.toml:33]: That's great news! Should we update the PR title or description to make this clear?

**Comment by sftim** [comment]
A nit: https://deploy-preview-29679--kubernetes-io-main-staging.netlify.app/hi/releases/ includes some unexpected English text:
“Complete” and “version skew policy”

File: config.toml
Comment (rolfedh) [reply config.toml:445]: I have closed my PR in favor of @xzyangNB's PR, https://github.com/kubernetes/website/pull/35898, which resolves this issue.

**Comment by rolfedh** [comment]
> A nit: https://deploy-preview-29679--kubernetes-io-main-staging.netlify.app/hi/releases/ includes some unexpected English text: “Complete” and “version skew policy”

Translated "version skew policy" with https://github.com/kubernetes/website/pull/35930

**Comment by rolfedh** [comment]
> A nit: https://deploy-preview-29679--kubernetes-io-main-staging.netlify.app/hi/releases/ includes some unexpected English text: “Complete” and “version skew policy”

The English portions of "Complete 1.24 [Schedule](https://deploy-preview-29679--kubernetes-io-main-staging.netlify.app/releases/patch-releases/#1-24) and [Changelog](https://git.k8s.io/kubernetes/CHANGELOG/CHANGELOG-1.24.md)" seem to be artifacts of the `{{< release-data >}}` shortcode.

**Comment by sftim** [comment]
> The English portions of "Complete 1.24 [Schedule](https://deploy-preview-29679--kubernetes-io-main-staging.netlify.app/releases/patch-releases/#1-24) and [Changelog](https://git.k8s.io/kubernetes/CHANGELOG/CHANGELOG-1.24.md)" seem to be artifacts of the `{{< release-data >}}` shortcode&hellip;

Ah, yes. We haven't got around to making that text localizable.


We could fix _version skew policy_ if we wish (doesn't block launching the localization).

**Comment by rolfedh** [comment]
> Ah, yes. We haven't got around to making that text localizable.
https://github.com/kubernetes/website/issues/35929

> We could fix _version skew policy_ if we wish (doesn't block launching the localization).
https://github.com/kubernetes/website/issues/35932

**Comment by Garima-Negi** [comment]
LGTM! looks ready to merge 👍

**Comment by Babapool** [comment]
The PR LGTM. Looking ready to merge now 👍 💯

**Comment by rolfedh** [comment]
- @Garima-Negi @Babapool would you provide `/lgtm` and `/approved` labels?
- Currently, Prow is not merging any PRs to kubernetes/website on branches `main` and `dev-1.25`. Refer to (#36171 and #36171) I expect this condition will end and this PR will merge when the merge freeze is over.

**Comment by sftim** [comment]
/hold
It's Kubernetes release day; I'd prefer _not_ to launch this localization at the point we unfreeze.

OK to unhold once a SIG Docs chair or tech lead is happy to launch the localization.

**Review by anubha-v-ardhan — COMMENTED** [review]
An LGTM from me!

**Comment by divya-mohan0209** [comment]
LGTM from me! We can proceed for this after the release, I think. I'd also want one of the other tech leads/co-chairs to chime in on this.

**Comment by rolfedh** [comment]
> LGTM from me! We can proceed for this after the release, I think. I'd also want one of the other tech leads/co-chairs to chime in on this.

@natalisucks or @reylejano WDYT?

**Comment by AmanSarraf** [comment]
great

**Comment by verma-kunal** [comment]
LGTM from me as well! 
Great work everyone 🎉

**Comment by sftim** [comment]
@rolfedh, did you have a plan to check (or help another person check) that _main_ will build OK if we merge in these changes?

**Comment by rolfedh** [comment]
> @rolfedh, did you have a plan to check (or help another person check) that _main_ will build OK if we merge in these changes?

I'll invite folks on the #kubernetes-docs-hi channel to walk through the local build and inspection process some time in the next few days.

**Comment by rolfedh** [comment]
TLDR: I have tested this merge and everything seems to work fine.

1. Merged this PR with `main` on my local machine and it merged cleanly.
2. I built the site using `make container-serve` and it built without errors.
3. Inspected the home page and various doc topics in various languages and everything looks fine.
4. For the non-Hindi localizations, I did "flipped" between the production and test site and saw only some very minor differences in spacing in the upper nav bar.

I believe this PR is ready to merge.
I think this

**Review by rolfedh — APPROVED** [review]

**Comment by rolfedh** [comment]
> /hold It's Kubernetes release day; I'd prefer _not_ to launch this localization at the point we unfreeze.
> 
> OK to unhold once a SIG Docs chair or tech lead is happy to launch the localization.

@sftim where does one look up this information?

**Comment by rolfedh** [comment]
@divya-mohan0209 @Garima-Negi @reylejano @sftim, would you remove the hold and approve this PR for merge?

**Comment by rolfedh** [comment]
Thanks @natalisucks! Now this PR needs an `/approved` label from someone else.

**Comment by anubha-v-ardhan** [comment]
Hindi localization is still working on `dev-1.22-hi.1`, so each time we merge a PR `lgtm` label is removed. I guess we'll need `/lgtm` and `/approved` at the same time 😅

**Comment by reylejano** [comment]
/hold
placing a hold until ready to merge into main

re-applying lgtm
/lgtm

/approve

feel free to unhold when ready to merge

**Comment by sftim** [comment]
Hindi approvers: if you need a new LGTM for this PR, and have confirmed that the changes since the last LGTM look OK, reply here and consider also asking in #sig-docs-maintainers on Slack.

This looks good to go. :tada:

**Pull request merged**
